site/content/blog/2019-12-18-cloud-monitoring.md

```diff
@@ -0,0 +1,91 @@
+
+{{< figure src="/blog/2019-12-18-cloud-monitoring/prometheus_federate.png" alt="Prometheus Federate">}}
+
+Both of the tools are deployed in a central command and control cluster using [Helm](https://helm.sh/) and [Terraform](https://www.terraform.io/). More information on the Terraform deployment code can be found [here](https://github.com/mattermost/mattermost-cloud-monitoring/tree/master/terraform/aws/modules/cluster-post-installation). For testing purposes the default version of Prometheus server can be installed by running:
+
+```bash
+helm install stable/prometheus
+```
+
+In order to install the Prometheus federation server, changes are required in the Helm values.yaml file and especially in the prometheus.yml section. The scrape_configs need to be replaced by a federate job name. For more information have a look in the code snippet below.
+
+```yaml
+prometheus.yml:
+    rule_files:
+      - /etc/config/rules
+      - /etc/config/alerts
+
+    scrape_configs:
+      - job_name: 'federate'
+        scrape_interval: 15s
+
+        honor_labels: true
+        metrics_path: '/federate'
+
+        params:
+          'match[]':
+            - '{job="prometheus"}'
+            - '{__name__=~"job:.*"}'
+
+        static_configs:
+          - targets:
+            - 'prometheus-server:80'
+```
+
+In this case we assume that the Prometheus federate config will be used only for data scraping from other Prometheus targets but another scrape config can be added in order to use the same Prometheus server for local cluster data scraping. We decided to not move forward with local data scraping as we would like to keep the Prometheus federation server dedicated to only one task and use another server for that.
+
+The next step in the setup is to specify the Prometheus server targets that the Prometheus federate server will use to scrape the data. For the targets, each server is registered with Route 53 and the static config looks like this:
+
+```yaml
+static_configs:
+      - targets:
+        - prometheus1.example.com.
+        labels:
+          clusterID: cluster1
+      - targets:
+        - prometheus2.example.com.
+        labels:
+          clusterID: cluster2
+```
+
+Using labels, makes the identification of the clusters easier in the Grafana dashboards and panels.
+
+Having the ability to add and remove targets, when a new Prometheus server is registered or deregistered with Route53 in an automated way is really important, which is why we developed a [Lambda function](https://github.com/mattermost/mattermost-cloud-monitoring/tree/master/prometheus-dns-registration-service) that handles the updates of the Prometheus server configmap, when a new Route53 record is created/deleted.
```

Comment:
Maybe talk about the lambda part in the DNS resolution section below?

Reply:
@jwilander The lambda is mostly related to the Prometheus itself because it just adds and removes Prometheus targets. Are you talking about another Lambda or about giving more details about this one?

Reply:
0/5 I guess I was thinking it was DNS resolution related when it's really not